# A look into the appstore_games.csv
pulling the data into variables first and getting a first look at them all

In [58]:
import pandas as pd
import numpy as np

In [66]:
df = pd.read_csv('appstore_games.csv')
ls_name = df['Name'].dropna()
ls_price = df['Price'].dropna()
ls_inapp = df['In-app Purchases'].dropna()
ls_avg_user_rating = df['Average User Rating'].dropna()

In [67]:
print(ls_name)

0                           Sudoku
1                          Reversi
2                          Morocco
3                    Sudoku (Free)
4                     Senet Deluxe
                   ...            
17002    Stack Puzzle : Rise Tower
17003                    EachOther
17004           Rabbit Vs Tortoise
17005                       FaTaLL
17006     The Three Kingdoms :Bomb
Name: Name, Length: 17007, dtype: object


In [68]:
print(ls_price)
np.average(ls_price)

0        2.99
1        1.99
2        0.00
3        0.00
4        2.99
         ... 
17002    0.00
17003    0.00
17004    0.00
17005    0.00
17006    0.00
Name: Price, Length: 16983, dtype: float64


0.8134187128304776

In [81]:
#print(ls_avg_user_rating[10])
np.mean(ls_avg_user_rating)

4.06090464224309

In [71]:
#print(ls_inapp)
for add in ls_inapp: 
    if add != None:
        print(add)

1.99
0.99
0.99
1.99, 0.99, 1.99, 0.99, 4.99, 1.99, 1.99
0.99, 0.99, 0.99
0.99, 0.99, 0.99, 0.99
1.99, 0.99
0.99, 2.99, 1.99
1.99
0.0, 0.99, 5.99, 0.99, 0.99, 0.99, 0.99, 0.99, 0.99, 0.99
2.99, 2.99, 5.99, 0.99, 0.0, 0.99, 0.99, 0.99, 0.99, 0.99
9.99, 9.99, 0.99, 0.99, 0.99, 0.99, 0.99, 0.99, 0.99, 0.99
0.99, 0.99, 0.99
0.99, 1.99, 0.99, 0.99
9.99, 2.99, 4.99, 19.99, 5.49, 23.49, 49.99, 36.99, 13.99, 99.99
0.99, 0.99, 0.99, 0.99, 0.99, 0.99, 0.99, 0.99, 0.99, 0.99
4.99, 0.99, 0.99, 0.99, 0.99, 0.99, 0.99, 0.99, 0.99, 0.99
0.99, 0.99
0.99, 0.99, 0.99, 1.99, 0.99, 2.99, 4.99, 0.99, 0.99, 0.99
1.99, 1.99, 1.99, 1.99, 1.99, 1.99, 1.99, 1.99, 5.99
0.99
1.99, 9.99, 0.99, 9.99, 2.99, 0.99, 19.99, 0.99, 0.99, 9.99
1.99, 7.99, 29.99, 19.99
0.99, 0.99
4.99, 9.99, 3.99, 4.99, 3.99, 9.99, 19.99, 49.99, 99.99
0.99
0.99, 5.99, 2.99
2.99
1.99
2.99, 1.99, 1.99, 1.99, 1.99, 1.99, 1.99, 1.99, 1.99, 1.99
19.99, 4.99, 9.99, 49.99, 99.99
0.99, 0.99, 0.99
0.99
0.99
0.99
4.99, 2.99, 2.99
99.99, 6.99, 14.99, 2

# A error came up due to gaps in the Data
I tried a couple of ways to go around it (shown in comments including above as a check) though it hadn't worked 

In [89]:
i = 0
hold = 0.0
while i < len(ls_name):
    if ls_price[i] == 0:  
        #if ls_avg_user_rating[i] != None:
        print(ls_avg_user_rating[i])
    i += 1    

3.0
3.5
3.0
2.5
2.5
2.5


KeyError: 10

In [92]:
zero_count = 0
for price in ls_price:
    if price == 0:
        zero_count += 1 
print(f'Price = 0: {zero_count} /(out of {len(ls_price)})')
print(np.average(ls_avg_user_rating))

Price = 0: 14212 /(out of 16983)
4.06090464224309


# a vast majority of games are made free to download:
overall the average rating is pretty high hovering around 4.1 stars